### PR TITLE
refactor(material/table): switch to tokens API

### PR DESCRIPTION
**Pull request by crisbeto**
Reworks the Material table to use the new tokens API.

### CODE REVIEW

File: src/material/table/_table-theme.scss [hunk 1]
```diff
@@ -1,60 +1,44 @@
...
-  @include mdc-helpers.using-mdc-theme($config) {
-    mdc-data-table-theme.$selected-row-fill-color: rgba(mdc-theme-color.prop-value(primary), 0.04);
-    mdc-data-table-theme.$divider-color: rgba(mdc-theme-color.prop-value(on-surface), 0.12);
-    mdc-data-table-theme.$row-hover-fill-color: rgba(mdc-theme-color.prop-value(on-surface), 0.04);
-    mdc-data-table-theme.$header-row-text-color: rgba(mdc-theme-color.prop-value(on-surface), 0.87);
-    mdc-data-table-theme.$row-text-color: rgba(mdc-theme-color.prop-value(on-surface), 0.87);
-    mdc-data-table-theme.$stroke-color: rgba(mdc-theme-color.prop-value(on-surface), 0.12);
-
-    @include mdc-data-table.table-styles($query: mdc-helpers.$mdc-theme-styles-query);
+@mixin _output-tokens {
```
Comment (crisbeto) [inline src/material/table/_table-theme.scss:6]: Will clean this up in a follow-up. It's a workaround for an internal issue.

File: src/material/table/table.scss [hunk 2]
```diff
@@ -25,6 +43,57 @@
...
   white-space: normal;
+
+  @include token-utils.use-tokens(tokens-mat-table.$prefix, tokens-mat-table.get-token-slots()) {
+    @include token-utils.create-token-slot(background-color, background-color);
+  }
+}
+
+@include mdc-helpers.disable-mdc-fallback-declarations {
+  @include token-utils.use-tokens(tokens-mat-table.$prefix, tokens-mat-table.get-token-slots()) {
+    .mat-mdc-header-row {
```
Comment (crisbeto) [inline src/material/table/table.scss:54]: I had to leave the fallback values for some tokens to reduce the amount of screenshot failures. Will try to clean them up in a follow up.

File: src/material/core/tokens/_token-utils.scss [hunk 3]
```diff
@@ -71,12 +71,21 @@ $_component-prefix: null;
 
 // Emits a slot for the given token, provided that it has a non-null value in the token map passed
 // to `use-tokens`.
-@mixin create-token-slot($property, $token) {
+@mixin create-token-slot($property, $token, $emit-fallback: false) {
```
Comment (andrewseguin) [inline src/material/core/tokens/_token-utils.scss:74]: Should we comment that having `emit-fallback = true` would be for temporary workarounds?

**Review by andrewseguin — APPROVED** [review]

File: src/material/table/_table-theme.scss (hunk 1 above)
Comment (mmalerba) [inline src/material/table/_table-theme.scss:6]: I think you can use this function instead: https://github.com/angular/components/blob/main/src/material/core/style/_sass-utils.scss#L11

Also, I'm not sure we need to clean it up, I'd like to move toward emitting the tokens under the user's selector (or document root if they don't specify one)

**Pull request merged**